File: tests/llvm/normal_blend_8.cc
#include "list.h"

std::vector<int> normal_blend_8(std::vector<int> base, std::vector<int> active, int opacity) 
{
  std::vector<int> out;
  for (int i = 0; i < base.size(); ++i)
    out.push_back(opacity * active[i] + (255 - opacity) * base[i]);

  return out;
}

File: tests/llvm/gaudi/benchmark/python/normal_blend_8.py
from typing import List


def normal_blend_8(base: List[int], active: List[int], opacity: int) -> List[int]:
    output: List[int] = []
    for i in range(len(base)):
        output.append(opacity * active[i] + (255 - opacity) * base[i])
    return output
